src/protocol.h

```diff
@@ -371,7 +371,13 @@ class CAddress : public CService
             READWRITE(nVersion);
         }
         if ((s.GetType() & SER_DISK) ||
-            (nVersion >= CADDR_TIME_VERSION && !(s.GetType() & SER_GETHASH))) {
+            (nVersion != INIT_PROTO_VERSION && !(s.GetType() & SER_GETHASH))) {
```

Comment:
style-nit: the python code reads much nicer, so I'd slightly prefer if the C++ code was also easy to read. Not blocking of course, but for example passing down an option `with_time` per https://github.com/bitcoin/bitcoin/issues/19477#issuecomment-656331810 would be nice.

Reply:
I'm working on that. In a branch I'm working on I have this patch that make another (vaguely related) readability improvement:

```patch
diff --git a/src/protocol.h b/src/protocol.h
index 985f44640b..b74a24e007 100644
--- a/src/protocol.h
+++ b/src/protocol.h
@@ -359,6 +359,11 @@ class CAddress : public CService
 {
     static constexpr uint32_t TIME_INIT{100000000};
 
+    /** The disk serialization for CAddress includes a version number.
+     *  Traditionally the number CLIENT_VERSION was used, but has since
+     *  been disentangled from it. */
+    static constexpr int DISK_VERSION{201000};
+
 public:
     CAddress() : CService{} {};
     explicit CAddress(CService ipIn, ServiceFlags nServicesIn) : CService{ipIn}, nServices{nServicesIn} {};
@@ -366,13 +371,14 @@ public:
     SERIALIZE_METHODS(CAddress, obj)
     {
         SER_READ(obj, obj.nTime = TIME_INIT);
-        int nVersion = s.GetVersion();
         if (s.GetType() & SER_DISK) {
-            READWRITE(nVersion);
-        }
-        if ((s.GetType() & SER_DISK) ||
-            (nVersion >= CADDR_TIME_VERSION && !(s.GetType() & SER_GETHASH))) {
+            int disk_version = DISK_VERSION;
+            READWRITE(disk_version);
             READWRITE(obj.nTime);
+        } else if (s.GetType() & SER_NETWORK) {
+            if (s.GetVersion() >= CADDR_TIME_VERSION) {
+                READWRITE(obj.nTime);
+            }
         }
         READWRITE(Using<CustomUintFormatter<8>>(obj.nServices));
         READWRITEAS(CService, obj);
```